### PR TITLE
Fix data location of external reference parameters.

### LINKED ISSUE #4771: External functions somehow require storage pointers
```
contract c {
    function f(bytes calldata ) external {}
    function g() public {
        bytes memory a;
        this.f(a);
    }
}
```
results in 
```
browser/Untitled12.sol:5:16: TypeError: Invalid type for argument in function call. Invalid implicit conversion from bytes memory to bytes storage pointer requested.
        this.f(a);
               ^
```

### PR DESCRIPTION
**Pull request by chriseth**
Fixes https://github.com/ethereum/solidity/issues/4771

This should fix our external tests.

### CODE REVIEW

**Comment by chriseth** [comment]
Yep, fixed.

**Comment by chriseth** [comment]
Codecov is really not reliable - it reports everything fine while obviously, the line in question is partly untested: https://codecov.io/gh/ethereum/solidity/pull/4790/src/libsolidity/analysis/ReferencesResolver.cpp#L347

File: libsolidity/analysis/ReferencesResolver.cpp [hunk 1]
```diff
@@ -344,7 +344,7 @@ void ReferencesResolver::endVisit(VariableDeclaration const& _variable)
 						"(remove the \"memory\" or \"storage\" keyword)."
 					);
 			}
-			if (varLoc == Location::Default)
+			if (varLoc == Location::Default || varLoc == Location::CallData)
 				typeLoc = DataLocation::CallData;
 			else
 				typeLoc = varLoc == Location::Memory ? DataLocation::Memory : DataLocation::Storage;
```
Comment (axic) [inline libsolidity/analysis/ReferencesResolver.cpp:350]: Why not move this into a switch statement instead?

Comment (chriseth) [reply libsolidity/analysis/ReferencesResolver.cpp:350]: because the code will be removed in https://github.com/ethereum/solidity/pull/4738

**Review by axic — APPROVED** [review]

**Pull request merged**